### ISSUE TITLE
Search error on settings: brain; message=null java.lang.NullPointerException

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#241]
This is caused by jena using a deprecated class in xercesImpl that was removed.

```
ERROR ubic.gemma.core.search.SearchServiceImpl.search(256) | Search error on settings: brain; message=null
java.lang.NullPointerException
        at ubic.basecode.ontology.model.OntologyTermImpl.getChildren(OntologyTermImpl.java:389)
        at ubic.basecode.ontology.model.OntologyTermImpl.getChildren(OntologyTermImpl.java:130)
        at ubic.gemma.core.search.SearchServiceImpl.getDirectChildTerms(SearchServiceImpl.java:2061)
        at ubic.gemma.core.search.SearchServiceImpl.getCharacteristicsAnnotatedToChildren(SearchServiceImpl.java:2011)
        at ubic.gemma.core.search.SearchServiceImpl.characteristicEESearchTerm(SearchServiceImpl.java:808)
        at ubic.gemma.core.search.SearchServiceImpl.characteristicEESearchWithChildren(SearchServiceImpl.java:903)
        at ubic.gemma.core.search.SearchServiceImpl.characteristicEESearch(SearchServiceImpl.java:715)
        at ubic.gemma.core.search.SearchServiceImpl.expressionExperimentSearch(SearchServiceImpl.java:1622)
        at ubic.gemma.core.search.SearchServiceImpl.accreteResultsOthers(SearchServiceImpl.java:447)
        at ubic.gemma.core.search.SearchServiceImpl.generalSearch(SearchServiceImpl.java:1880)
        at ubic.gemma.core.search.SearchServiceImpl.search(SearchServiceImpl.java:297)
        at ubic.gemma.core.search.SearchServiceImpl.search(SearchServiceImpl.java:250)
        at sun.reflect.GeneratedMethodAccessor1357.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
        at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:69)
        at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:255)
        at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:148)
        at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:90)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
        at com.sun.proxy.$Proxy118.search(Unknown Source)
        at ubic.gemma.web.controller.GeneralSearchControllerImpl.ajaxSearch(GeneralSearchControllerImpl.java:116)
        at sun.reflect.GeneratedMethodAccessor1373.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:166)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
        at com.sun.proxy.$Proxy412.ajaxSearch(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor1372.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
```

**Comment by arteymix** [comment]
Fixed in f82b3dd0a033eb7b9e529ee40a8053de8dfe7f43

**Comment by arteymix** [comment]
There's also this issue with jena that will be fixed as well:

```
ERROR ubic.gemma.core.ontology.providers.GeneOntologyServiceImpl$1.run(838) | java.lang.NoClassDefFoundError: org/apache/xerces/impl/dv/xs/AbstractDateTimeDV$DateTimeData
java.lang.NoClassDefFoundError: org/apache/xerces/impl/dv/xs/AbstractDateTimeDV$DateTimeData
        at org.apache.xerces.impl.dv.xs.DateTimeDV.getActualValue(Unknown Source)
        at org.apache.xerces.impl.dv.xs.XSSimpleTypeDecl.getActualValue(Unknown Source)
        at org.apache.xerces.impl.dv.xs.XSSimpleTypeDecl.validate(Unknown Source)
        at com.hp.hpl.jena.datatypes.xsd.XSDDatatype.parse(XSDDatatype.java:272)
        at com.hp.hpl.jena.graph.impl.LiteralLabelImpl.setValue(LiteralLabelImpl.java:213)
        at com.hp.hpl.jena.graph.impl.LiteralLabelImpl.setLiteralLabel_1(LiteralLabelImpl.java:107)
        at com.hp.hpl.jena.graph.impl.LiteralLabelImpl.<init>(LiteralLabelImpl.java:96)
        at com.hp.hpl.jena.graph.impl.LiteralLabelFactory.createLiteralLabel(LiteralLabelFactory.java:28)
        at com.hp.hpl.jena.graph.Node.createLiteral(Node.java:101)
        at com.hp.hpl.jena.rdf.arp.JenaReader.convert(JenaReader.java:120)
        at com.hp.hpl.jena.rdf.arp.JenaReader.convert(JenaReader.java:143)
        at com.hp.hpl.jena.rdf.arp.JenaHandler.statement(JenaHandler.java:89)
        at com.hp.hpl.jena.rdf.arp.impl.XMLHandler.triple(XMLHandler.java:86)
        at com.hp.hpl.jena.rdf.arp.impl.ParserSupport.triple(ParserSupport.java:228)
        at com.hp.hpl.jena.rdf.arp.states.WantDescription.aPredAndObj(WantDescription.java:116)
        at com.hp.hpl.jena.rdf.arp.states.WantPropertyElement.theObject(WantPropertyElement.java:213)
        at com.hp.hpl.jena.rdf.arp.states.WantTypedLiteral.endElement(WantTypedLiteral.java:50)
        at com.hp.hpl.jena.rdf.arp.impl.XMLHandler.endElement(XMLHandler.java:133)
        at org.apache.xerces.parsers.AbstractSAXParser.endElement(Unknown Source)
        at org.apache.xerces.impl.XMLNamespaceBinder.handleEndElement(Unknown Source)
        at org.apache.xerces.impl.XMLNamespaceBinder.endElement(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
        at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at org.apache.xerces.parsers.DTDConfiguration.parse(Unknown Source)
        at org.apache.xerces.parsers.DTDConfiguration.parse(Unknown Source)
        at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
        at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
        at com.hp.hpl.jena.rdf.arp.impl.RDFXMLParser.parse(RDFXMLParser.java:155)
        at com.hp.hpl.jena.rdf.arp.JenaReader.read(JenaReader.java:173)
        at com.hp.hpl.jena.rdf.arp.JenaReader.read(JenaReader.java:160)
        at com.hp.hpl.jena.rdf.arp.JenaReader.read(JenaReader.java:199)
        at com.hp.hpl.jena.rdf.model.impl.ModelCom.read(ModelCom.java:229)
        at com.hp.hpl.jena.ontology.impl.OntModelImpl.read(OntModelImpl.java:2126)
        at ubic.basecode.ontology.OntologyLoader.loadMemoryModel(OntologyLoader.java:246)
        at ubic.basecode.ontology.OntologyLoader.loadMemoryModel(OntologyLoader.java:183)
        at ubic.gemma.core.ontology.providers.GeneOntologyServiceImpl.loadTermsInNameSpace(GeneOntologyServiceImpl.java:628)
        at ubic.gemma.core.ontology.providers.GeneOntologyServiceImpl$1.run(GeneOntologyServiceImpl.java:827)
        at java.lang.Thread.run(Thread.java:748)
```
